File: cpp/74-Search-a-2D-Matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int r=matrix.size();
        int c=matrix[0].size();
        for(int i=0;i<r;i++)
        {
            
           if(target>=matrix[i][0] && target<=matrix[i][c-1])
               if(binary_search(matrix[i].begin(),matrix[i].end(),target))
               return true;
            else return false;
           
                
        }
        return false;
    }
};

File: python/74-Search-a-2D-Matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        ROWS = len(matrix)

        for i in range(ROWS):
            if target > matrix[i][-1] or target < matrix[i][0]:
                continue
            else:
                arr = matrix[i]
                l, r = 0, len(arr) - 1
                while l <= r:
                    m = (l + r) // 2
                    if target > arr[m]:
                        l = m + 1
                    elif target < arr[m]:
                        r = m - 1
                    else:
                        return True
        return False